src/crimson/os/seastore/cache.cc

```diff
@@ -1043,7 +1045,9 @@ CachedExtentRef Cache::duplicate_for_write(
   if (i->is_exist_clean()) {
     i->version++;
     i->state = CachedExtent::extent_state_t::EXIST_MUTATION_PENDING;
-    i->last_committed_crc = i->get_crc32c();
+    auto crc = i->calc_crc32c();
+    i->set_last_committed_crc(crc);
+    i->update_in_extent_chksum_field(crc);
```

Comment:
Seems it's unrelated to logical and can be removed.

Reply:
Done.

Reply:
>ERROR 2024-04-18 03:50:57,863 [shard 0:main] none - /home/jenkins-build/build/workspace/ceph-pull-requests/src/crimson/os/seastore/cache.cc:1884 : In function 'auto crimson::os::seastore::Cache::replay_delta(crimson::os::seastore::journal_seq_t, crimson::os::seastore::paddr_t, const crimson::os::seastore::delta_info_t &, const crimson::os::seastore::journal_seq_t &, const crimson::os::seastore::journal_seq_t &, crimson::os::seastore::sea_time_point)::(anonymous class)::operator()(auto) const [extent:auto = boost::intrusive_ptr<crimson::os::seastore::CachedExtent>]', ceph_assert(%s)
extent->last_committed_crc == delta.prev_crc
Aborting on shard 0.

It seems that we have to set last_committed_crc() here, because, later when preparing records, deltas need to keep the correct "prev_crc" in themselves. What do you think?

Reply:
I left this line untouched, which should be fine now.